src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -7154,118 +7215,41 @@ std::pair<MergeTreeData::MutableDataPartPtr, scope_guard> MergeTreeData::cloneAn
     bool copy_instead_of_hardlink,
     const NameSet & files_to_copy_instead_of_hardlinks)
 {
-    /// Check that the storage policy contains the disk where the src_part is located.
```

Comment:
This method has been extracted into the `MergeTreeDataPartCloner` class in order to re-use its code with `MergeTreeDataPartDistinctPartitionExpressionCloner`.